lib/puppet/settings.rb

```diff
@@ -1342,7 +1342,8 @@ def convert(value, setting_name)
           interpolated_expression
         end
         if failed_environment_interpolation
-          Puppet.warning("You cannot interpolate $environment within '#{setting_name}' when using directory environments.  Its value will remain #{interpolated_value}.")
+          Puppet.warning(_("You cannot interpolate $environment within '%{setting_name}' when using directory environments.") % { setting_name: setting_name } +
```

Comment:
I'm trying to figure out what's going on with `$environment` here. Is that a variable that the user sets? Is it required to be `environment`, or can that variable have a different name? I have a feeling that shouldn't be translated, but I'd like to get confirmation on my interpretation. Maybe @joshcooper or @mwaggett can weigh in?

Reply:
@melissa Did you mean to ping me?

Reply:
Oops, no! Apologies for that! I meant @Magisus !

Reply:
I don't really know, but I have the same feeling, it probably shouldn't be translated. My guess is it's referring to the `$environment` setting, which has a fixed name.